src/chips/add_carry_v1.rs

```diff
@@ -0,0 +1,162 @@
+use halo2_proofs::{circuit::*, halo2curves::pasta::Fp, plonk::*, poly::Rotation};
+
+#[derive(Debug, Clone)]
+pub struct AddCarryConfig {
+    pub advice: [Column<Advice>; 3],
+    pub instance: Column<Instance>,
+    pub selector: Selector,
+}
+
+#[derive(Debug, Clone)]
+pub struct AddCarryChip {
+    config: AddCarryConfig,
+}
+
+impl AddCarryChip {
+    pub fn construct(config: AddCarryConfig) -> Self {
+        Self { config }
+    }
+
+    pub fn configure(
+        meta: &mut ConstraintSystem<Fp>,
+        advice: [Column<Advice>; 3],
+        selector: Selector,
+        instance: Column<Instance>,
+    ) -> AddCarryConfig {
+        let col_a = advice[0];
+        let col_b = advice[1];
+        let col_c = advice[2];
+        let add_carry_selector = selector;
+
+        // Enable equality on the advice and instance column to enable permutation check
+        meta.enable_equality(col_a);
+        meta.enable_equality(col_b);
+        meta.enable_equality(col_c);
+        meta.enable_equality(instance);
+
+        // enforce dummy hash function by creating a custom gate
+        meta.create_gate("accumulate constraint", |meta| {
+            let s = meta.query_selector(add_carry_selector);
+            let prev_b = meta.query_advice(col_b, Rotation::prev());
+            let prev_c = meta.query_advice(col_c, Rotation::prev());
+            let a = meta.query_advice(col_a, Rotation::cur());
+            let b = meta.query_advice(col_b, Rotation::cur());
+            let c = meta.query_advice(col_c, Rotation::cur());
+
+            // Previous accumulator amount + new value from a_cell
+            // using binary expression (x_n-4 * 2^16) + (x_n-3 * 2^8) + ... + (x_n * 2)
+            vec![
+                s * ((a + (prev_b * Expression::Constant(Fp::from(1 << 16))) + prev_c)
+                    - ((b * Expression::Constant(Fp::from(1 << 16))) + c)),
+            ]
+        });
+
+        AddCarryConfig {
+            advice: [col_a, col_b, col_c],
+            instance,
+            selector: add_carry_selector,
+        }
+    }
+
+    // Initial accumulator values from instance for expreiment
+    pub fn assign_first_row(
```

Comment:
I suggest adding the first row using the `Value<Fp>` as input for the `assign_first_row` function  and then decomposing it in 16_bits chunks and assigning it to the `acc` columns inside the function. 

By removing the need to copy the value from the instance column, the chip becomes more reusable for all the circuits that do not want the value to be public in the first place